pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -537,7 +537,7 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 		destroyInfrastructure = g.Add(flow.Task{
 			Name:         "Destroying shoot infrastructure",
 			Fn:           flow.TaskFn(botanist.Shoot.Components.Extensions.Infrastructure.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout),
-			Dependencies: flow.NewTaskIDs(syncPointCleaned, waitUntilControlPlaneDeleted),
+			Dependencies: flow.NewTaskIDs(syncPointCleaned, waitUntilControlPlaneDeleted, waitUntilShootManagedResourcesDeleted),
```

Comment:
Why is the `Infrastructure` destruction task now dependent on this? Shouldn't it be the `kube-apiserver` deletion task? 👀

Reply:
Good question. Now that I have given it a second thought this dependency seems like it is not needed.